packages/filesystem/src/browser/location/location-renderer.tsx

```diff
@@ -14,14 +14,14 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { h } from '@phosphor/virtualdom';
-import { VirtualRenderer } from "@theia/core/lib/browser";
 import URI from "@theia/core/lib/common/uri";
 import { LocationService } from "./location-service";
+import { ReactRenderer } from "@theia/core/lib/browser/widgets/react-renderer";
+import * as React from 'react';
 
 export const LOCATION_LIST_CLASS = 'theia-LocationList';
 
-export class LocationListRenderer extends VirtualRenderer {
+export class LocationListRenderer extends ReactRenderer {
```

Comment:
please make sure that `LocationListRenderer` is disposed together with `FileDialog`

Reply:
https://github.com/theia-ide/theia/pull/2301/files#diff-f62deb5fbabbd0bb37a733cc83c2ba8fR86